QuizApp/App/Screens/Quiz/QuizViewController.swift

```diff
@@ -15,6 +26,70 @@ final class QuizViewController: UIViewController {
         createViews()
         styleViews()
         defineLayoutForViews()
+        setupDelegateAndDataSource()
+    }
+
+}
+
+// MARK: - UICollectionViewDataSource methods
+extension QuizViewController: UICollectionViewDataSource {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == categoryCollectionView {
+            return quizViewModel.quizCategories.count
+        } else {
+            return 4
+        }
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        if collectionView == categoryCollectionView {
+            guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: CategoryCell.reuseIdentifier,
+                for: indexPath
+            ) as? CategoryCell else { fatalError() }
+
+            cell.data = Category(name: quizViewModel.quizCategories[indexPath.item].name)
+
+            if indexPath.item == 0 {
```

Comment:
Why are you autoselecting an item every time this is update? If this is so that the app autoloads the first category, this should be done on VM level and init.